### PR TITLE
Fixes #1424 - Album support deezer

### LINKED ISSUE #1424: Add album/playlist search to Deezer
#1293 added album and playlist support to Spotify and YouTube, it could also be added to Deezer

### PR DESCRIPTION
**Pull request by melvin2016**
**Outreachy applicants:** please try to get all _new_ PRs submitted by Friday,
April 3rd. After that we'll be trying to review all PRs and respond to 
followups. Earlier is better, as the review load is very high.

Before submitting a final PR, please:

- [ ] Run `npm test` on your machine
- [x] Run `npm run format` to keep code formatted consistently
- [x] Reference the issue you are fixing in at least one of your commit messages, as `Fixes #X`

Fixes #1424 .

### CODE REVIEW

**Comment by melvin2016** [comment]
@ianb Please review :)

File: extension/services/deezer/player.js [hunk 1]
```diff
@@ -1,52 +1,68 @@
...
+          }
+        } catch (e) {
+          throw new Error("No Search Results!");
         }
       } catch (e) {
-        const unlogged = this.querySelector("div[class='unlogged-homepage']");
-        if (unlogged) {
-          throw new Error("Please log in to use this service.");
-        }
+        throw new Error(e.message);
```
Comment (ianb) [inline extension/services/deezer/player.js:39]: This doesn't do anything any longer (except break the traceback by recreating the exception). But it also looks like this code isn't right. It should probably be:

```js
...} catch (e) {
  const unlogged = document.querySelector(".unlogged-homepage");
  if (unlogged) {
    throw new Error("Please log in to use this service.");
  }
  throw e;
}
```

File: extension/services/deezer/player.js [hunk 2]
```diff
@@ -124,6 +140,39 @@ this.player = (function() {
         iconVolumeParent.click();
       }
     }
+
+    async playSection({ query, thenPlay, section }) {
+      try {
+        let SECTION;
```
Comment (ianb) [inline extension/services/deezer/player.js:146]: This shouldn't be an upper class variable, that should only be used for constants or global config variables, not a mutable local variable. Maybe `foundSection` would be a good name?

File: extension/services/deezer/player.js [hunk 3]
```diff
@@ -124,6 +140,39 @@ this.player = (function() {
...
+        await this.search(query);
+        try {
+          if (thenPlay) {
+            const anchorNodes = await this.waitForSelector(".container h2 a", {
+              all: true,
+              timeout: 5000,
+            });
+            for (const anchorTag of anchorNodes) {
+              if (anchorTag.innerText === section) {
+                SECTION = anchorTag.parentElement.parentElement;
```
Comment (ianb) [inline extension/services/deezer/player.js:156]: This should probably `break;` once it is found

File: extension/services/deezer/player.js [hunk 4]
```diff
@@ -124,6 +140,39 @@ this.player = (function() {
...
+              }
+            }
+            SECTION.querySelector("ul li button").click();
```
Comment (ianb) [inline extension/services/deezer/player.js:159]: This needs to check if `SECTION` is still undefined. Probably it should just raise an exception saying that the section couldn't be found.

File: extension/services/deezer/player.js (hunk 3 above)
Comment (melvin2016) [reply extension/services/deezer/player.js:156]: Oh. Sorry. Yes it was taking much time searching for all the sections thus making it slow.

File: extension/services/deezer/player.js (hunk 4 above)
Comment (melvin2016) [reply extension/services/deezer/player.js]: Ok

File: extension/services/deezer/player.js (hunk 1 above)
Comment (ianb) [reply extension/services/deezer/player.js:43]: If it always re-throws the error, then the try/catch isn't necessary at all.

**Comment by melvin2016** [comment]
> Looks good, thanks! There's one issue with try/catch that I'll just fix in the interest of getting this landed.

Thank You! :)

**Pull request merged**